### ISSUE TITLE
Personal youtube API key gets overwritten

### ISSUE THREAD

**Issue by malick-2892** [anxdpanic/plugin.video.youtube#214]
### Context

Please provide any relevant information about your setup

* Add-on Version: 6.8.17
* Kodi Version: 18.9.0 (Leia)
* Kodi GUI Language: English
* Operating System: LibreELEC 9.2.8 (RPi2.arm)
* Operating System Language: english

------

### Expected Behavior

Youtube url passed from either webbrowser plugin or android app (weather official youtube or alternative android app - NewPipe, Yatse, Kore) would check against stored personal credentials (api key, client id, client secret), then using these would proceed to acquire videostream list and pass appropriate stream to kodi to play. I have MPEG-DASH enabled and configured. 

------

### Current Behavior

Instead, the credentials (api_keys.json) get overwritten (by default ones, hardcoded?) and used instead, which results in either stream buffering (throttled by youtube I guess), reduced stream quality (e.g. 1080p feed plays at 144p or 240p hic!) or stream terminated. The plugin had been working properly for me previously, being configured. Then I noticed video quality reduction, stream stuttering and buffering, occasionally stream being bounced on request, hence I started to troubleshoot the issue. The issue must've occurred first about 2 weeks ago I think. My troubleshooting ended up discovering my YT API credentials got overwritten by the plugin.

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Goto plugin configuration-> API -> provide credentials
2. Pass a youtube url that has 720p+ stream (preferably MPEG-DASH) , for instance:
https://www.youtube.com/watch?v=M9hXQW8k-Io
3. Once stream starts playing check video stream details. In this RobboNuggie's footage top available stream is 1080p50, but once chosen, whether automatically or manually, the supposed 1080p50 downgrades to 360p. Should 720p (at 25fps) stream be chosen video stuttering, buffering and basically throttle occurs.
4. Check API credentials, and magically those aren't own personal key(s)/id/secrets any more.

------

### Log

[kodi.zip](https://github.com/anxdpanic/plugin.video.youtube/files/7338427/kodi.zip)

------

### Additional Information

n/a

------

**Comment by claes** [comment]
I don't have any credentials registered in the addon, but the part where the stream resolution is downgraded I also experience now. For example, if I create a .strm file referring to a Youtube video, or sharing the above Youtube URL to Kodi via Yatse.
I think it started within the last 48-24 hours. 
My addon version is 6.8.17+matrix.1

**Comment by malick-2892** [comment]
@claes 
If You don't have your YT personal api credentials (key, id, secret)  configured with the plugin You go with te default one which results in such behaviour. At some point in time, several months ago original developer's key(s) wouldn't work at all as those had been overloaded (over default quota). For that reason users have been advised to create and use personal keys, which I've followed.

**Comment by malick-2892** [comment]
Update:
I have removed plugin, reinstalled v6.8.15, which worked fine for me previously as fas as I can recall, put my YT credentials.
Those stick and change no more. Than I bumped up to 6.8.17 and it's cool. Must've messed up with config somewhere on the road. :)
That's so much for good news.
I must've been red flagged by YT for whatever reason cause any stream attempt (whether mpd or mpg) ends up in either stream downgrade (mpd via use mpeg-dash automatically select adaptive mpd) or massive throttle that brings my user experience to it's knees, being able to pull @ ~ 60kbps :<
